### ISSUE TITLE
Add 2016 election precinct shapes for New Hampshire

### ISSUE THREAD

**Issue by nvkelso** [nvkelso/election-geodata#42]
Priority 4 based on population

**Comment by nvkelso** [comment]
When researching & requesting data also ask for **matching election results** (_presidential, congressional, state house, and state senate_) in addition to the precinct shape data for 2016, **and for earlier election years** if available.

**Comment by wboykinm** [comment]
Also dibs.

**Comment by nvkelso** [comment]
- Voting Districts, New Hampshire, 2000
    - http://geocommons.com/datasets?id=5137
    - http://www.census.gov/geo/www/cob/vt2000.html
    - https://raw.githubusercontent.com/geoiq/gc_data/master/datasets/5137.geojson

**Comment by wboykinm** [comment]
The [geocommons-hosted districts](http://geocommons.com/datasets?id=5137) suffer from the same problem the Vermont districts ( #48) had; the rural areas of NH don't vote at the county level, but rather the town. I'll see if there's anything like a polygon boundary available.

**Comment by nvkelso** [comment]
Can you clarify, please? Are votes cast at the town (or minor civil division) level only, there aren't separate precincts for reporting purposes?

**Comment by wboykinm** [comment]
Sure, votes are collected and reported at the "district" level, which is more or less analogous with "precinct". The full list of these districts is [here](https://www.dropbox.com/s/81brnzdwwvkpdo1/TOWNS%20AND%20WARDS%20AS%20DISTRICTED%20FOR%20ELECTION%20PURPOSES2015.pdf?dl=0), and election results (at least since 2012) have been [reported at that level](http://sos.nh.gov/2012PresGen.aspx?id=28307).

The complication (similar to Vermont) is that no geometric representation of the complete boundaries seems to exist. NH-GRANIT hosts a ["political boundaries" layer](http://www.granit.unh.edu/data/metadata?file=pb/nh/pb.html) that conforms to the upper level of the SoS reporting districts, but it's missing ward subdivisions that exist in all cities and constitute the highest level of detail the SoS reports.

Ward boundaries exist _somewhere_ (see [this example for Lebanon,NH](http://lebcity.net/CityClerk/Documents/Election/wardmap.pdf)), but neither GRANIT nor the SoS host them, making this potentially a city-by-city slog. 

@nvkelso While I try to get some clarification from GRANIT, do you think it's worth importing the coarser political boundaries that *are* available?

**Comment by nvkelso** [comment]
Yes, we should import the coarser political boundaries and then supplement
them with the more detailed wards later.

There are two examples on how to do this now:

- Kansas: see the Makefile and associated script
- Indiana: pure Makefile

Basically we can cookie cutter pieces of the more detailed wards into the
more generic districts for a composite state output. If you upload the
statewide districts I can take care of doing the integration work once the
wards start landing (and they can land 1 by 1 as you find them with new
PRs).

**Comment by migurski** [comment]
This does suggest that our output schema should include a flag for division level: precincts and wards are a lot different from larger house districts.

**Comment by wboykinm** [comment]
I AM LEARNING SO MUCH ABOUT DEMOOCRAZY

**Comment by nvkelso** [comment]
Hahahaha. Me too :)

Bill can you clarify what "district" means in your case? I assume it's
something smaller than a "congressional district", but am not familiar with
how New England sets things up.

**Comment by wboykinm** [comment]
"Voting District" refers to the catchment within which Granite Staters vote. One polling place per district, so it includes both towns and wards.

**Comment by wboykinm** [comment]
@nvkelso You [already added an exact version of what we're looking for!](https://github.com/nvkelso/election-geodata/commit/b4f7f77c96b602153b8f17e1b6774d5896c873e4) These are just outdated since the 2012 redistricting. Did you get them from the census?

**Comment by nvkelso** [comment]
Yep, Census 2010.

**Comment by wboykinm** [comment]
I suspect that means someone at the census went through what we're doing right now back in 2009, and isn't scheduled to do so again until 2019 😬

**Comment by nvkelso** [comment]
Yep :\ But you have very small states ;)

**Comment by wboykinm** [comment]
### Ward maps (PDF unless otherwise noted)
- [Lebanon](http://lebcity.net/CityClerk/Documents/Election/wardmap.pdf)
- [Manchester](https://www.manchesternh.gov/Portals/2/Departments/city_clerk/wards/CitywideWardMap.pdf)
- [Dover](https://dovernh.mapgeo.io/?latlng=43.17561%2C-70.885208&panel=themes&themes=%5B%22wards-and-polling-places%22%5D&zoom=12) (possibly-scrapable Gmaps app)
- [Concord](http://nh-concord.civicplus.com/DocumentCenter/View/1201) (Note the unironic use of comic sans)
- [Franklin](http://www.franklinnh.org/Pages/FranklinNH_Council/map.pdf)
- [Keene](http://www.ci.keene.nh.us/sites/default/files/WARDS_12_GRID_0.pdf)
- [Laconia](https://www.google.com/maps/d/u/0/viewer?mid=1oqNwYVdgUhjL5ZhPRlY-XxaEujs&ie=UTF8&hl=en&msa=0&ll=43.55004100000002%2C-71.47035599999998&spn=0.084851%2C0.137501&z=13) (very-scrapable google mymaps page)
- [Nashua](http://www.nashuanh.gov/DocumentCenter/View/3836)
- [Portsmouth](http://www.cityofportsmouth.com/cityclerk/documents/votingWards.pdf)
- [Rochester](http://www.rochesternh.net/sites/rochesternh/files/file/file/rochester_wards_06-2016.pdf)
- [Somersworth](http://www.somersworth.com/wp-content/uploads/2012/11/Somersworth-Wards.pdf)

@nvkelso This is all of the cities in NH with ward subdivisions.

**Comment by nvkelso** [comment]
These are great, thanks!

**Comment by nvkelso** [comment]
I think what's going on here is _most_ of the state is still current in 2016 using the 2010 US Census data **except** the ~11 places mentioned in https://github.com/nvkelso/election-geodata/issues/42#issuecomment-286913534. I think the Makefile is correct, but these areas are pretty small so the state looks entirely green in the preview map – but if you look closer there are some gaps.

**Comment by wboykinm** [comment]
@nvkelso This is my next project 😀
